importing sqlite3 fails in dev mode

### Describe the bug

A recent update to sveltekit broke the usage of sqlite3 in server-side code.

When using npm's sqlite3 module in server-side code and running the code in development mode, the import fails with

```
Cannot find module 'node_modules/sqlite3/package.json' imported from 'node_modules/node-pre-gyp/lib/pre-binding.js'
    at viteResolve (node_modules/vite/dist/node/chunks/dep-4a9cff06.js:67808:25)
```

### Reproduction

use sqlite3 in a svelte hook, and run the code with `svelte-kit dev`

Reproducible in https://github.com/lovasoa/sanipasse/tree/5f4a3d5177c88462b2bee2c029eeb7534155cdb2

### Logs

```shell
Error: Cannot find module '/home/olojkine/dev/sanipasse/node_modules/sqlite3/package.json' imported from '/home/olojkine/dev/sanipasse/node_modules/node-pre-gyp/lib/pre-binding.js'
    at viteResolve (/home/olojkine/dev/sanipasse/node_modules/vite/dist/node/chunks/dep-4a9cff06.js:67808:25)
    at Function._resolveFilename (/home/olojkine/dev/sanipasse/node_modules/vite/dist/node/chunks/dep-4a9cff06.js:67822:20)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.exports.find (/home/olojkine/dev/sanipasse/node_modules/node-pre-gyp/lib/pre-binding.js:20:23)
    at Object.<anonymous> (/home/olojkine/dev/sanipasse/node_modules/sqlite3/lib/sqlite3-binding.js:3:27)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
```


### System Info

```shell
System:
    OS: Linux 5.15 Fedora Linux 35 (Workstation Edition)
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 1.18 GB / 15.28 GB
    Container: Yes
    Shell: 5.1.8 - /bin/bash
  Binaries:
    Node: 16.13.0 - /usr/bin/node
    Yarn: 1.22.10 - /usr/bin/yarn
    npm: 8.1.0 - /usr/bin/npm
  Browsers:
    Firefox: 95.0
  npmPackages:
    @sveltejs/adapter-node: next => 1.0.0-next.56 
    @sveltejs/adapter-static: next => 1.0.0-next.22 
    @sveltejs/kit: next => 1.0.0-next.202 
    svelte: ^3.38.3 => 3.44.3 

```


### Severity

blocking an upgrade

### Additional Information

_No response_

--- a/packages/playground/ssr-deps/package.json
+++ b/packages/playground/ssr-deps/package.json
@@ -17,6 +17,7 @@
     "only-object-assigned-exports": "file:./only-object-assigned-exports",
     "primitive-export": "file:./primitive-export",
     "read-file-content": "file:./read-file-content",
+    "require-absolute": "file:./require-absolute",
     "ts-transpiled-exports": "file:./ts-transpiled-exports"
   },
   "devDependencies": {

--- a/packages/playground/ssr-deps/require-absolute/foo.js
+++ b/packages/playground/ssr-deps/require-absolute/foo.js
@@ -1,0 +1,1 @@
+module.exports.hello = 'Hello World!'

--- a/packages/playground/ssr-deps/require-absolute/index.js
+++ b/packages/playground/ssr-deps/require-absolute/index.js
@@ -1,0 +1,3 @@
+const path = require('path')
+
+module.exports.hello = () => require(path.resolve(__dirname, './foo.js')).hello

--- a/packages/playground/ssr-deps/require-absolute/package.json
+++ b/packages/playground/ssr-deps/require-absolute/package.json
@@ -1,0 +1,5 @@
+{
+  "name": "require-absolute",
+  "version": "0.0.0",
+  "private": true
+}

--- a/packages/playground/ssr-deps/src/app.js
+++ b/packages/playground/ssr-deps/src/app.js
@@ -8,6 +8,7 @@ import bcrypt from 'bcrypt'
 import definePropertiesExports from 'define-properties-exports'
 import definePropertyExports from 'define-property-exports'
 import onlyObjectAssignedExports from 'only-object-assigned-exports'
+import requireAbsolute from 'require-absolute'
 
 export async function render(url, rootDir) {
   let html = ''
@@ -40,6 +41,9 @@ export async function render(url, rootDir) {
 
   const onlyObjectAssignedExportsMessage = onlyObjectAssignedExports.hello()
   html += `\n<p class="only-object-assigned-exports-msg">message from only-object-assigned-exports: ${onlyObjectAssignedExportsMessage}</p>`
+
+  const requireAbsoluteMessage = requireAbsolute.hello()
+  html += `\n<p class="require-absolute-msg">message from require-absolute: ${requireAbsoluteMessage}</p>`
 
   return html + '\n'
 }

--- a/packages/vite/src/node/ssr/ssrModuleLoader.ts
+++ b/packages/vite/src/node/ssr/ssrModuleLoader.ts
@@ -2,6 +2,7 @@ import path from 'path'
 import { pathToFileURL } from 'url'
 import type { ViteDevServer } from '../server'
 import {
+  bareImportRE,
   dynamicImport,
   isBuiltin,
   unwrapId,
@@ -232,14 +233,9 @@ async function nodeImport(
   // When an ESM module imports an ESM dependency, this hook is *not* used.
   const unhookNodeResolve = hookNodeResolve(
     (nodeResolve) => (id, parent, isMain, options) => {
-      // Fix #5709, use require to resolve files with the '.node' file extension.
-      // See detail, https://nodejs.org/api/addons.html#addons_loading_addons_using_require
-      if (
-        id[0] === '.' ||
-        path.isAbsolute(id) ||
-        isBuiltin(id) ||
-        id.endsWith('.node')
-      ) {
+      // Use the Vite resolver only for bare imports while skipping
+      // any built-in modules and binary modules.
+      if (!bareImportRE.test(id) || isBuiltin(id) || id.endsWith('.node')) {
         return nodeResolve(id, parent, isMain, options)
       }
       if (parent) {

--- a/packages/vite/src/node/ssr/ssrModuleLoader.ts
+++ b/packages/vite/src/node/ssr/ssrModuleLoader.ts
@@ -234,7 +234,12 @@ async function nodeImport(
     (nodeResolve) => (id, parent, isMain, options) => {
       // Fix #5709, use require to resolve files with the '.node' file extension.
       // See detail, https://nodejs.org/api/addons.html#addons_loading_addons_using_require
-      if (id[0] === '.' || isBuiltin(id) || id.endsWith('.node')) {
+      if (
+        id[0] === '.' ||
+        path.isAbsolute(id) ||
+        isBuiltin(id) ||
+        id.endsWith('.node')
+      ) {
         return nodeResolve(id, parent, isMain, options)
       }
       if (parent) {

